### ISSUE TITLE
When trying to execute a POST, the first time consistently fails

### ISSUE THREAD

**Issue by rob-deans** [IBM-Swift/Kitura#699]
Every subsequent one runs after the first failure.

Code that I am running: 

```
router.post("/login") {
    request, response, next in
    
    //Get current session
    sess = request.session
    
    var maybeEmail: String?
    
    switch request.body {
    case .urlEncoded(let params)?:
        maybeEmail = params["email"]
    case .json(let params)?:
        maybeEmail = params["email"].string
    default: break
    }
    
    if let email = maybeEmail, let sess = sess {
        sess["email"] = JSON(email)
        try response.send("done").end()
    }
}
```

**Comment by ianpartridge** [comment]
Can you publish a GitHub project that someone could clone to easily reproduce this issue?

**Comment by youming-lin** [comment]
I tested this locally with this code on Kitura-0.28.0:

```swift
import HeliumLogger
import Kitura
import KituraSession
import LoggerAPI
import SwiftyJSON

Log.logger = HeliumLogger()

let router = Router()
let session = Session(secret: "pizza")

router.all(middleware: session)
router.all(middleware: BodyParser())

router.all(handler: {request, response, next in
    Log.info(request.url)
    next()
})

router.post("/login") { request, response, next in
    //Get current session
    let sess = request.session

    var maybeEmail: String?

    switch request.body {
    case .urlEncoded(let params)?:
        maybeEmail = params["email"]
    case .json(let params)?:
        maybeEmail = params["email"].string
    default: break
    }

    if let email = maybeEmail, let sess = sess {
        sess["email"] = JSON(email)
        try response.send("done").end()
    }
}

Kitura.addHTTPServer(onPort: 3000, with: router)
Kitura.run()
```

I've tested this using curl: `curl -v -X POST -H "Content-Type: application/json" -d '{"email": "user@example.com"}' localhost:3000/login`, and I get 200 OK even on the first hit.

@rob-deans Can you list your reproduction steps?

**Comment by rob-deans** [comment]
@youming-lin  when doing curl as well, it works fine, however when doing it through all the major browsers is when it is failing, I have made a GitHub project here: https://github.com/rob-deans/Kitura-Session-Test , the problem is the body is `nil` on the first run through of `router.post("/login")`

**Comment by youming-lin** [comment]
I see, so this is somewhat of an opposite of #692.

**Comment by youming-lin** [comment]
@rob-deans I noticed that your test code is on Kitura-0.26.0. Can you update it to Kitura-0.28.0 and try again? Per #692 a fix to that was delivered to 0.28.

**Comment by rob-deans** [comment]
@youming-lin Yes the update to Kitura-0.28.0 has fixed the issue
